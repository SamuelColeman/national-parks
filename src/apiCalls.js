export const fetchParks = async (state) => {
	const response = await fetch(`https://developer.nps.gov/api/v1/parks?stateCode=${state}&api_key=jd03rGVMgfyl9RFBbSTUwXPh1IBNhbUf6Mt40p3E`);
	const data = await response.json();
	console.log(data)
	return data.data;
}