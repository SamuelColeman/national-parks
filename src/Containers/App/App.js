import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './App.css';
import { fetchParkInfo, fetchParks } from '../../apiCalls';
import { hasError, getInfo, getInfoName, getParks, isLoading } from '../../actions';
import { Route } from 'react-router-dom';
import ParksContainer from '../ParksContainer/ParksContainer';
import Form from '../Form/Form';
import ParkCard from '../ParkCard/ParkCard';
import InfoContainer from '../InfoContainer/InfoContainer';

export class App extends Component {

  submitState = async (state) => {
    const { hasError, getParks, isLoading } = this.props;
    try {
      isLoading(true);
      const parks = await fetchParks(state);
        isLoading(false);
      if (parks.length > 0) {
        getParks(parks);
        hasError('');
      } else {
        hasError('Invalid State');
      }
    } catch (error) {
      hasError(error.message);
    }
  }

  displayParkInfo = async (code, e) => {
    const { hasError, getInfo, getInfoName } = this.props;
    try {
      getInfoName(e.target.name);
      const info = await fetchParkInfo(code, e.target.name);
      getInfo(info);
    } catch (error) {
      hasError(error.message);
    }
  }

  handleSearch = (e) => {
    const { parks, getParks } = this.props;
    let upperSearch = e.target.value.toUpperCase();
    let searchParks = parks.filter(park => {
       return (park.name.toUpperCase().includes(upperSearch))
    });
    getParks(searchParks);
  }

  render() {
    const { infoName } = this.props;
    return (
      <div className="App">
        <Route exact path='/parks' render={() => <ParksContainer displayParkInfo={this.displayParkInfo} handleSearch={this.handleSearch} submitState={this.submitState} />} />
        <Route exact path='/' render={() => <Form submitState={this.submitState}/>} />
        <Route exact path='/parks/:id' render={({ match }) => {
            let { parks } = this.props;
            const { id } = match.params;
            const matchPark = parks.find(park => park.id === id);
            return (
              <ParkCard {...matchPark} page={true} displayParkInfo={this.displayParkInfo}/>
          )}} />
        <Route exact path={`/parks/:id/${infoName}`} render={() => 
          <InfoContainer />} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  parks: state.parks,
  infoName: state.infoName
});

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    hasError,
    getInfo,
    getInfoName,
    getParks,
    isLoading
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App);