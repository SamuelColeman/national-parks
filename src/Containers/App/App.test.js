import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';

import { fetchParks, fetchParkInfo } from '../../apiCalls';


jest.mock('../../apiCalls');

describe('App', () => {

	let wrapper;
	let mockParks = [{ 
    	description: "Historic Site",
			designation: "National Historic Site",
			directionsInfo: "Drive Car",
			directionsUrl: "http://www.directions.com",
			fullName: "National Historic Site",
			id: "2",
			latLong: "lat:36, long:82",
			name: "Historic Park",
			parkCode: "pant",
			states: "TN",
			url: "https://www.example.com",
			weatherInfo: "Sunny"
		}];
	let mockParkInfo = [{
    	description: "Biggest visitor center ever.",
			directionsInfo: "Take a left.",
			directionsUrl: "https://www.exampledirections.com",
			id: "11",
			latLong: "{lat:35, lng:83}",
			name: "Visitor Center",
			parkCode: "pant",
			url: "https://www.examplecenter.com"
		}];
	let mockInfoName = 'VisitorCenters';
	let mockState = 'TN';
	let mockIsLoading = jest.fn();
	let mockGetParks = jest.fn();

	fetchParks.mockImplementation(() => {
        return Promise.resolve(mockParks);
      }); 
  fetchParkInfo.mockImplementation(() => {
        return Promise.resolve(mockParkInfo)
      })

	beforeEach(() => {
    wrapper = shallow(<App parks={mockParks} isLoading={mockIsLoading} getParks={mockGetParks} infoName={mockInfoName}/>)
 	})

  it('should match snapshot with correct data passing through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call isLoading when submitState is invoked', async () => {
  	wrapper.instance().submitState(mockState);
  	expect(mockIsLoading).toHaveBeenCalledWith(true);
  	expect(fetchParks).toHaveBeenCalledWith(mockState);
  })
});