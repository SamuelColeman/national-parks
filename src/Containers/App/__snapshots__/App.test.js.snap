// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with correct data passing through 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/parks"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/parks/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/parks/:id/VisitorCenters"
    render={[Function]}
  />
</div>
`;
